start

```diff
@@ -512,6 +512,11 @@ function upgrade_local() {
   # Wait for server
   while ! echo "Stellar-core http server listening!" | nc localhost 11626 &> /dev/null; do sleep 1; done
 
+  # Default to latest version supported by core if no default or explicit version was set
+  if [ -z "$PROTOCOL_VERSION" ]; then
+    export PROTOCOL_VERSION=`curl -s http://localhost:11626/info | jq -r '.info.protocol_version'`
```

Comment:
is there any race condition potential on core's http `/info` endpoint not being ready to respond yet? just wondering if it warrants an error correction/retry?

Reply:
The line above waits for core's http to be available. The lines I'm adding here used to be here for a very long time and were only recently removed. So I don't know for sure if there could never be a race, but these lines of code did their things successfully for years before without noticeable problems so I think we can plop them back and see how it goes again.

Reply:
sounds good, thanks!